### ISSUE TITLE
Wrong code at -O2/3/s on x86_64-linux-gnu since clang-20

### ISSUE THREAD

**Issue by XChy** [llvm/llvm-project#203868]
Reproducer: https://godbolt.org/z/KrsxP8MrT
Testcase:
```c
#include <stdint.h>
#include <stdio.h>
int64_t __chk, g28, g19;
_Bool launch_c0;
int main()
{
    int64_t arr1[9];
    _Bool c3, c6;
    int64_t ov11;
lbl_br1:
    ov11 = g28;
    c3 = c6 = 0;
    if (c6) goto lbl_br1;
    c6 = (int32_t)&c3;
    g19 = 0;
    if (c3) goto lbl_bf7;
lbl_br6:
    if (c6)
    {
        c6 = 8;
    lbl_bf7:
        arr1[4] = ov11;
        if (c6)
        {
            c6 = 0;
            ov11 = arr1[0];
        }
        launch_c0;
        goto lbl_br6;
    }
    __chk = arr1[4];
    printf("checksum=0x%016llx\n", (unsigned long long)__chk);
    return 0;
}
```

This testcase outputs `checksum=0x0000000000000000` at -O0, and outputs `checksum=0x00007ffd707deb38` at -O2/3/s instead.

**Comment by pektezol** [comment]
Isn't this not a valid miscompile due to UB?

**Comment by XChy** [comment]
> Isn't this not a valid miscompile due to UB?

I reduced the original testcase to this one with ASan, MSan and UBSan, though they have some false negatives...

It looks like `ov11 = arr1[0]` uses the uninitialized value, but this indeterminate value should not be propagated to the program output. I manually initialize `arr1[0]` with `123`, and the output is still different between -O0 and -O2: https://godbolt.org/z/Wa8PT9WWv

**Comment by lijinpei-amd** [comment]
To me, `JumpThreadingPass::duplicateCondBranchOnPHIIntoPred` is the one to blame, but I'm still investigating on a proper fix.

**Comment by XChy** [comment]
> To me, `JumpThreadingPass::duplicateCondBranchOnPHIIntoPred` is the one to blame, but I'm still investigating on a proper fix.

May be related to https://github.com/llvm/llvm-project/pull/198443?

**Comment by lijinpei-amd** [comment]
> > To me, `JumpThreadingPass::duplicateCondBranchOnPHIIntoPred` is the one to blame, but I'm still investigating on a proper fix.
> 
> May be related to [#198443](https://github.com/llvm/llvm-project/pull/198443)?

Yes, #198443 can fix it. And I'm working on a fix without rejecting the transformation, by cloning the phi into a dedicated BB, it can also fix the two issues, but I'm still polishing the fix https://github.com/lijinpei-amd/llvm-project/tree/2026-06-19-jumpthreading-dup-phi-miscompile
